src/platform/plugins/shared/controls/common/transforms/data_controls/references.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { omit } from 'lodash';
+
+import type { Reference } from '@kbn/content-management-utils';
+import type { DataControlState } from '@kbn/controls-schemas';
+import { DATA_VIEW_SAVED_OBJECT_TYPE } from '@kbn/data-views-plugin/common';
+import type { StoredDataControlState } from './types';
+
+export function extractReferences(
+  state: DataControlState,
+  referenceName: string
+): { state: StoredDataControlState; references?: Reference[] } {
+  return {
+    state: {
+      ...omit(state, 'dataViewId'),
+      dataViewRefName: referenceName,
+    },
+    references: [
+      {
+        name: referenceName,
+        type: DATA_VIEW_SAVED_OBJECT_TYPE,
+        id: state.dataViewId,
+      },
+    ],
+  };
+}
+
+export function injectReferences(
```

Comment:
It doesn't look like references are getting injected correctly when a control is added to a section. 

<img width="1104" height="692" alt="Image" src="https://github.com/user-attachments/assets/23244159-6de6-45c3-a3fc-625f445ffdb6" />

<img width="445" height="306" alt="Image" src="https://github.com/user-attachments/assets/d73c5c94-befd-41e7-87ee-c98d0538f3d5" />